### ISSUE TITLE
Deep_Archive not a storage class option in S3

### ISSUE THREAD

**Issue by bradstapleton94** [aws/aws-cli#5133]
Confirm by changing [ ] to [x] below:
- [x] I've gone though the [User Guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) and the [API reference](https://docs.aws.amazon.com/cli/latest/reference/)
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-cli/issues) and didn't find any solution

Issue is about usage on:
- [ ] Service API : I want to do X using Y service, what should I do?
- [x] CLI : passing arguments or cli configurations.
- [ ] Other/Not sure.

**Platform/OS/Hardware/Device**

Raspbian GNU/Linux 10 (buster)

**Describe the question**
I am attempting to copy files using the AWS CLI to S3 using the DEEP_ARCHIVE storage class as outlined in this doc: https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html?highlight=storage%20class.

The CLI is responding that DEEP_ARCHIVE is not a storage class option.

aws: error: argument --storage-class: Invalid choice, valid choices are:

STANDARD                                 | REDUCED_REDUNDANCY
STANDARD_IA                              | ONEZONE_IA
INTELLIGENT_TIERING                      | GLACIER

**Logs/output**
Get full traceback and error logs by adding `--debug` to the command.

2020-04-19 18:03:36,368 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.16.113 Python/3.7.3 Linux/4.19.66-v7+ botocore/1.12.103
2020-04-19 18:03:36,369 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['s3', 'cp', 'test.zip', 's3://cloudarchive-deep', '--storage-class', 'DEEP_ARCHIVE', '--sse', 'aws:kms', '--debug']
2020-04-19 18:03:36,371 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x751e3a50>
2020-04-19 18:03:36,371 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function register_uri_param_handler at 0x7557d6f0>
2020-04-19 18:03:36,373 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x7551a6f0>
2020-04-19 18:03:36,380 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x75299228>
2020-04-19 18:03:36,381 - MainThread - botocore.hooks - DEBUG - Event building-command-table.s3: calling handler <function add_waiters at 0x751ec7c8>
2020-04-19 18:03:36,386 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.custom.s3.anonymous: calling handler <awscli.paramfile.URIArgumentHandler object at 0x751bd3f0>
2020-04-19 18:03:36,388 - MainThread - botocore.hooks - DEBUG - Event building-command-table.cp: calling handler <function add_waiters at 0x751ec7c8>
usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help
aws: error: argument --storage-class: Invalid choice, valid choices are:

STANDARD                                 | REDUCED_REDUNDANCY
STANDARD_IA                              | ONEZONE_IA
INTELLIGENT_TIERING                      | GLACIER

**Comment by KaibaLopez** [comment]
Hi @bradstapleton94 ,
that's weird, for me the cli command is accepting deep_archive as a storage class and it works as expected. This was the command I used:
`aws s3 cp s3://your-bucket/your-obj s3://other-bucket/your-obj --storage-class DEEP_ARCHIVE"`
Can you share what the command that gave you the error was?

**Comment by bradstapleton94** [comment]
Thanks for the reply @KaibaLopez.  I just tried the below command and I was successful.  I did it a second time to make sure I was not crazy and it worked as well.  I did have to create a new access-key inside of IAM.  Not sure if that had something to do with it or not.

Is there a limit to the file type or size that can be uploaded via the CLI?  The files I was testing before were larger zip and tar.gz files.  I wouldn't think file type would matter but size?  I know glacier was a max of 4GB.  Larger files had to do multipart uploads with tree hashing.  Thanks for your continued guidance and support on this.

Command that worked: aws s3 cp inventory.json s3://BUCKET_NAME_HERE --storage-class DEEP_ARCHIVE --sse aws:kms
